### ISSUE TITLE
Plugin binding for `onServerStop` event

### ISSUE THREAD

**Issue by junzhengca** [kafuuchino-desu/MCDaemon#5]
`onServerStop` can be quite useful, for example, you might want to close the server and do an automatic tarball backup to AWS S3.

It might be possible to use `onServerInfo` to achieve similar functionality as `onServerStop`, but that is quite hacky, a consistent API is more desirable.

**Comment by kafuuchino-desu** [comment]
yes its already in our developement plan, still, thanks a lot for your issue, and btw not only onServerStop will be added into pluginsV2,a lot more apis will be added such as `onPluginReload` etc
